File: src/main/scala/com/emofy/OpenApiConfig.scala
package com.emofy

import org.springframework.context.annotation.{Bean, Configuration}
import org.springdoc.core.GroupedOpenApi

@Configuration
class OpenApiConfig {

  @Bean
  def publicApi(): GroupedOpenApi = {
    GroupedOpenApi.builder()
      .group("public-api")
      .pathsToMatch("/public/**")
      .build()
  }

  @Bean
  def privateApi(): GroupedOpenApi = {
    GroupedOpenApi.builder()
      .group("private-api")
      .pathsToMatch("/private/**")
      .build()
  }
}

File: login/src/main/java/com/example/demo/OpenApiConfig.java
package com.example.demo;
/* OLD CONFIGURATION
import springfox.documentation.swagger2.annotations.EnableSwagger2;
import org.springframework.context.annotation.Configuration;
import springfox.documentation.spring.web.plugins.Docket;
import org.springframework.context.annotation.Bean;
import springfox.documentation.spi.DocumentationType;
import springfox.documentation.builders.RequestHandlerSelectors;
import springfox.documentation.builders.PathSelectors;

@Configuration
@EnableSwagger2
public class AppConfig {

    @Bean
    public Docket api() {
        return new Docket(DocumentationType.SWAGGER_2)
            .select()
            .apis(RequestHandlerSelectors.any())
            .paths(PathSelectors.any())
            .build();
    }

}
*/
import org.springframework.context.annotation.Configuration;
import org.springdoc.core.GroupedOpenApi;
import org.springframework.context.annotation.Bean;

@Configuration
public class OpenApiConfig {

    @Bean
    public GroupedOpenApi publicApi() {
        return GroupedOpenApi.builder()
            .group("public-api")
            .pathsToMatch("/public/**")
            .build();
    }

    @Bean
    public GroupedOpenApi privateApi() {
        return GroupedOpenApi.builder()
            .group("private-api")
            .pathsToMatch("/private/**")
            .build();
    }
}
